hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MutableRatesWithAggregation.java

```diff
@@ -71,11 +74,20 @@ public void init(Class<?> protocol) {
     if (protocolCache.contains(protocol)) {
       return;
     }
-    protocolCache.add(protocol);
-    for (Method method : protocol.getDeclaredMethods()) {
-      String name = method.getName();
-      LOG.debug(name);
-      addMetricIfNotExists(name);
+
+    List<Class<?>> protocols = new ArrayList<>();
+    protocols.add(protocol);
+    if (protocol.getDeclaredMethods().length == 0) {
```

Comment:
make add some comments for this since it's not very obvious why we do it.

Reply:
ok. I'll add.